tests/v2/Controller/DocumentController/MoveToFolderTest.php

```diff
@@ -77,27 +77,29 @@ public function testMoveToWrongBeneficiary(): void
 
     public function provideTestMoveToFolder(): ?\Generator
     {
-        yield 'Shared document should be hydrated with parent visibility' => [false];
-        yield 'Private document should be hydrated with parent visibility' => [true];
+        yield 'Shared document moved in shared folder should be shared' => [false, false, false];
+        yield 'Shared document moved in private folder should be private' => [false, true, true];
+        yield 'Private document moved in shared folder should be private' => [true, false, true];
+        yield 'Private document moved in private folder should be private' => [true, true, true];
     }
 
     /** @dataProvider provideTestMoveToFolder */
-    public function testMoveToFolder(bool $isPrivate): void
+    public function testMoveToFolder(bool $isPrivateDoc, bool $isPrivateFolder, bool $result): void
```

Comment:
Maybe we could rename $result --> `$shouldBePrivate` or `$expectedPrivate`
Not really important